dev-lang/rust: add upstream fix from pr65932

stefson opened the pull request:
the bootstrap script uses python to extract the stage0, and given that
someone out there's still using python2 for that, it will fail, as there's
no built in support for lzma prior to python-3.4

the updated patch solves this problem, as it detects the used python version
and chooses tar.gz for users of python2 and tar.xz for those of python3

cnd commented:
@stefson I can't merge not signed off pull requests anymore after gentoo rules updates you need to add sign off (-s commit option)

cnd commented:
use can use `git commit --amend` to add sign off line

stefson commented:
Which rules have been updated? The last thing I read about this was, that all commits have to be signed-off by the author for the portage tree, and all overlays which use the gentoo git infrastructure (cgit.gentoo.git) 

In any case, I added the sign-off and updated the upstream patch.

cnd commented:
@stefson all projects moving out from github, that one is mirror to https://gitweb.gentoo.org/repo/proj/rust.git/

related: https://bugs.gentoo.org/698358

merged! thank you

stefson commented:
duh, what's going to happen with the modus operandi then if this github repo here is deleted?

cnd commented:
@stefson mirrors on github will be removed, and I will keep taking Pull Requests here
